File: src/main/java/net/kuryshev/webnotes/TaskRepository.java
package net.kuryshev.webnotes;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface TaskRepository extends JpaRepository<Task, Long> {

    List<Task> findAllByUsername(String username);

    Optional<Task> getByIdAndUsername(long id, String username);

    void deleteByIdAndUsername(long id, String username);
}

File: src/main/java/net/kuryshev/webnotes/TaskRepository.kt
package net.kuryshev.webnotes

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TaskRepository : JpaRepository<Task?, Long?> {
    fun findAllByUsername(username: String): List<Task>
    fun getByIdAndUsername(id: Long, username: String): Task?
    fun deleteByIdAndUsername(id: Long, username: String)
}